src/ruby.ts

```diff
@@ -32,200 +46,199 @@ export class RubyProcess extends EventEmitter {
 	public constructor(args: LaunchRequestArguments) {
 		super();
 		this.launchArgs = args;
+		this.pendingResponses = [];
 		this.pendingCommands = [];
+		this.socketConnected = false;
 		this.state = SocketClientState.ready;
 
-		var that = this;
 		var runtimeArgs = ['--evaluation-timeout', '10'];
 		var runtimeExecutable: string;
 
 		if (process.platform === 'win32') {
 			runtimeExecutable = 'rdebug-ide.bat';
 		}
-		else if (process.platform === 'darwin') {
-			runtimeExecutable = 'rdebug-ide';
-		}
 		else {
 			// platform: linux
 			runtimeExecutable = 'rdebug-ide';
 		}
 
-		var programArgs = [];
 		var processCwd = dirname(this.launchArgs.program);
+		//xd only if showDebuggerOutput is set - it is also a runtime arg, not a program arg
```

Comment:
FYI, the command ran in RubyMine is `/usr/bin/ruby -e at_exit{sleep(1)};$stdout.sync=true;$stderr.sync=true;load($0=ARGV.shift) /Users/rebornix/.gem/ruby/2.0.0/gems/ruby-debug-ide-0.4.32/bin/rdebug-ide --disable-int-handler --evaluation-timeout 10 --rubymine-protocol-extensions --port 55354 --dispatcher-port 55355 -- /Users/rebornix/Documents/code/testrb/test.rb` , I suppose we might use Ruby runtime someday, but for now let's stick with rdebug-ide.